crates/sncast/src/lib.rs

```diff
@@ -278,22 +279,36 @@ pub async fn wait_for_tx(
     }
     let retries = timeout / u16::from(retry_interval);
     for i in (1..retries).rev() {
-        match provider.get_transaction_receipt(tx_hash).await {
-            Ok(receipt) => {
-                return match receipt.execution_result() {
-                    ExecutionResult::Succeeded => Ok("Transaction accepted"),
-                    ExecutionResult::Reverted { reason } => {
-                        Err(anyhow!("Transaction has been reverted = {reason}"))
+        match provider.get_transaction_status(tx_hash).await {
+            Ok(status) => match status {
+                starknet::core::types::TransactionStatus::Received => {
+                    println!("Transaction has been received");
+                }
+                starknet::core::types::TransactionStatus::Rejected => {
+                    return Err(anyhow!("Transaction has been rejected"));
+                }
+                starknet::core::types::TransactionStatus::AcceptedOnL2(execution_status)
+                | starknet::core::types::TransactionStatus::AcceptedOnL1(execution_status) => {
+                    match execution_status {
+                        starknet::core::types::TransactionExecutionStatus::Succeeded => {
+                            return Ok("Transaction accepted")
+                        }
+
+                        starknet::core::types::TransactionExecutionStatus::Reverted => {
+                            return Err(anyhow!("Transaction has been reverted"))
```

Comment:
It could be done by calling rpc for a receipt. Status endpoint returns only status

Reply:
then maybe we should call for receipt in this place and give the user revert reason? we do not have a command to allow users to fetch it, that's why I think it might make sense. Could be removed after we add such functionality.